docs/proposals/20220812-daemon-service.md

```diff
@@ -0,0 +1,38 @@
+# Daemon Process Support
+
+Author:
+- [kemingy](https://github.com/kemingy/)
+
+
+## Summary
+
+This proposal is designed to support run daemon processes in the `envd` container.
+
+## Motivation
+
+There can be a general solution for several use cases:
+
+1. run a Jupyter notebook service
+2. run a TensorBoard service
+3. run a demo machine learning model serving service
+4. run a metrics exporter
+
+This is related to the following:
+
+* https://github.com/tensorchord/envd/issues/527
+* https://github.com/tensorchord/envd/pull/568
+* https://github.com/tensorchord/envd/pull/708
+* https://github.com/tensorchord/envd/pull/497
+
+## Goals
+
+* able to run multiple daemon processes controlled by `tini`
```

Comment:
What is the implementation plan? Any architectural considerations that we should discuss here?

Reply:
* plan: can add the commands to `tini` https://github.com/tensorchord/envd/blob/b9f0af8056129fe6c5a2e590cd428463186cac0e/pkg/lang/ir/compile.go#L166-L176
* I'd like to discuss if this is a general approach (need to work with other features like `mount` and `expose`) to solving the issues like #527 . Or if we should do it in another way?